handout/architect/10_ways_to_misuse_angular.md

```diff
@@ -0,0 +1,439 @@
+
+If you would like to provide a service as an application-level singleton use the `forRoot()` function. For providing services to feature modules you should always use `forChild()`.
+
+
+
+
+## 5. Confusing the Source of Truth Between Router and Application State
+The angular router allows us to define a client side routing structure and manages transitions between those routes. It does not however, manage or mutate your application state. You are responsible for updating the application state in response to navigation events.
+
+Angular router exposes routing events as an observable – `Router.events`. To synchronize your application state or to hydrate state on first load (based on the url) you should subscribe to the router events observable and dispatch suitable actions. Since the router events are the only hooks available to us we should ensure that all route transitions are triggered through `routerLink` or programatically using `router.navigate()`
+
+
+
+
+## 6. Not using content projection
+For components prefer projection over templates with complex nested markup. This allows you apply the principles of function composition for views. Instead of having one massive component which accounts for various scenarios you create a toolbox of smaller single responsibility components. This approach has several benefits:
+
+- It makes the component hierarchy flatter.
+- Avoids excessive chaining of `@Input`/`@Output` to pass state down/events up the component tree.
+- Testing components becomes much easier since you don’t have to deal with view logic.
+- Allows you to leverage existing components for implementing new features.
+
+This pattern is especially important for form elements since it allows us to connect models and event handlers at the container level instead of having to pass them down through multiple levels.
+
+### Example
+By creating generic wrapper components we can use the same set of components to create:
+- An image card
+- A Card with title and caption
+- A card with title, caption and an icon
+- and many more combinations
+
+```html
+<Card>
+  <Media [source]="'img/rotary-phone.png'"></Media>
+
+  <Block paddingX="2" paddingY="2">
+
+    <Heading size="2">
+      Choosing the Right Antique Rotary Phone for You.
+    </Heading>
+
+    <Block>
+      <Heading size="3" muted>21 hours ago</Heading>
+      <Button style="clear"
+        (handleClick)="addToCart($event)"
+      >
+        <Icon name="add-to-cart" />
+      </Button>
+    </Block>
+
+  </Block>
+</Card>
+```
+
+For more on this subject see the [Flattening Deep Hierarchies of Components](http://blog.rangle.io/flattening-deep-hierarchies-of-components) post on the Rangle.io blog.
+
+
+
+
+## 7. Using model-driven forms straight away
```

Comment:
@winkerVSbecks I think just focusing on the main points here would be sufficient. The outline for this part may have been a little... over-thought. Let me know if you'd like some help with this, but here's mainly what I was envisioning for this section.

> - Template-driven forms require less boilerplate, setup, and mental overhead

This can play off of the previous section about content projection, mention that you don't need to configure any "back-end" model, you can simply compose the form in your template.
Easy composition of these template-driven forms is facilitated by leveraging content projection so that form logic and state can be handled at the container level.

> Model-driven forms more appropriate for working with complex dynamically generated forms

This might best be expressed through an example, but really the point I was hoping to make is when ngIf's and ngFor's aren't enough, or become unreadable and unmaintainable, then you've probably ventured into territory when model-driven forms are a more appropriate solution.

I don't know if @smithad15 would have anything to add, but I think those are the main points